kolibri/plugins/hooks.py

```diff
@@ -291,7 +291,7 @@ def add_hook_to_class_registry(cls, hook):
             and cls._registered_hooks
             and hook not in cls.registered_hooks
         ):
-            raise RuntimeError(
+            logger.error(
```

Comment:
FYI @rtibbles I switched this because we were running into this issue: 

https://learningequality.slack.com/archives/CR0LK9W80/p1583968360115800?thread_ts=1583967561.115300&cid=CR0LK9W80

However I think now it's triggering this:

```
=================================== FAILURES ===================================
___________________ test_register_multiple_hook_single_hook ____________________
valid_single_hook = <class 'test.kolibri_plugin.SingleHook'>
    def test_register_multiple_hook_single_hook(valid_single_hook):
        valid_single_hook = hooks.register_hook(valid_single_hook)
        valid_single_hook.add_hook_to_registries()
    
        class OtherSingleHook(SingleHookAbstract):
            pass
    
        OtherSingleHook.__module__ = "test.kolibri_plugin"
        try:
            OtherSingleHook = hooks.register_hook(OtherSingleHook)
            OtherSingleHook.add_hook_to_registries()
            pytest.fail(
>               "Allowed a hook single instance hook to be registered more than once"
            )
E           Failed: Allowed a hook single instance hook to be registered more than once
kolibri/plugins/utils/test/test_hooks.py:80: Failed
----------------------------- Captured stderr call -----------------------------
ERROR    Attempted to register more than one instance of <class 'test.kolibri_plugin.OtherSingleHook'>
```

curious about your thoughts

Reply:
The issue was that we were not catching all errors that could occur during plugin initialization when trying to disable a plugin, which left the user in a catch 22. I've now shifted Kolibri to do a more 'self healing' kind of approach, and just aggressively disabling plugins that don't initialize, while writing loud error messages about this. Let me know if this is too far in the other direction.